app/src/main/java/org/simple/clinic/user/UserSession.kt

```diff
@@ -108,56 +107,15 @@ class UserSession @Inject constructor(
 
   fun register(): Single<RegistrationResult> {
     val user: Single<User> = loggedInUser()
-        .map { (user) -> user!! }
+        .filterAndUnwrapJust()
         .firstOrError()
         .cache()
 
     val currentFacility = user
-        .flatMap { facilityRepository.facilityUuidsForUser(it).firstOrError() }
```

Comment:
Curious, did the earlier case address the situation where a user may belong to multiple facilities? Trying to understand why this list of `UUIDs` were replaced with just the current facility?

Reply:
Ah, this is leftover from very old code. Initially, we left it to the user to decide which facility they wanted to be a part of so they would be select multiple facilities and we would send them all to the server.

However, this was later replaced by the server controlled facility group feature. The user would always register at only one facility and they would automatically get access to all facilities in that group. The api contracts were also changed to reflect this so the user registration api would accept a single facility ID instead of a list of facility IDs.

However, the local code still has some of this leftover behaviour (like being able to get a list of facility IDs for a user) which will always be a list of only one ID in reality, and we always use `.first()` or something to get the user's facility.

This goes one step towards removing that older code.